homeassistant/components/zwave_js/climate.py

```diff
@@ -0,0 +1,305 @@
+    client: ZwaveClient = hass.data[DOMAIN][config_entry.entry_id][DATA_CLIENT]
+
+    @callback
+    def async_add_climate(info: ZwaveDiscoveryInfo) -> None:
+        """Add Z-Wave Climate."""
+        entities: List[ZWaveBaseEntity] = []
+        entities.append(ZWaveClimate(client, info))
+
+        async_add_entities(entities)
+
+    hass.data[DOMAIN][config_entry.entry_id][DATA_UNSUBSCRIBE].append(
+        async_dispatcher_connect(
+            hass, f"{DOMAIN}_add_{CLIMATE_DOMAIN}", async_add_climate
+        )
+    )
+
+
+class ZWaveClimate(ZWaveBaseEntity, ClimateEntity):
+    """Representation of a Z-Wave climate."""
+
+    def __init__(self, client: ZwaveClient, info: ZwaveDiscoveryInfo) -> None:
+        """Initialize lock."""
+        super().__init__(client, info)
+        self._hvac_modes: Dict[str, Optional[int]] = {}
+        self._hvac_presets: Dict[str, Optional[int]] = {}
+        for enum in ThermostatSetpointType:
+            self.get_zwave_value(
+                THERMOSTAT_SETPOINT_PROPERTY,
+                command_class=CommandClass.THERMOSTAT_SETPOINT,
+                value_property_key_name=enum.value,
+                add_to_watched_value_ids=True,
+            )
+        self.get_zwave_value(
+            THERMOSTAT_OPERATING_STATE_PROPERTY,
+            command_class=CommandClass.THERMOSTAT_OPERATING_STATE,
+            add_to_watched_value_ids=True,
+        )
+        self.get_zwave_value(
+            THERMOSTAT_CURRENT_TEMP_PROPERTY,
+            command_class=CommandClass.SENSOR_MULTILEVEL,
+            add_to_watched_value_ids=True,
+        )
+
+        self.on_value_update()
+
+    @callback
+    def on_value_update(self) -> None:
+        """Call when a watched value is added or updated."""
+        self._current_mode_setpoint_values = self._get_current_mode_setpoint_values()
+        if not self._hvac_modes:
+            self._set_modes_and_presets()
+
+    def _get_current_mode_setpoint_values(self) -> List[ZwaveValue]:
+        """Return a list of current setpoint Z-Wave value(s)."""
+        current_mode = int(self.info.primary_value.value)
+        setpoint_enums: List[ThermostatSetpointType] = THERMOSTAT_MODE_SETPOINT_MAP.get(
+            current_mode, []
+        )
+        # we do not want None values in our list so check if the value exists
+        values = []
+        for enum in setpoint_enums:
+            value: Optional[ZwaveValue] = self.get_zwave_value(
+                THERMOSTAT_SETPOINT_PROPERTY,
+                CommandClass.THERMOSTAT_SETPOINT,
+                value_property_key_name=enum.value,
+            )
+            if value:
+                values.append(value)
+        return values
+
+    def _set_modes_and_presets(self) -> None:
+        """Convert Z-Wave Thermostat modes into Home Assistant modes and presets."""
+        all_modes: Dict[str, Optional[int]] = {}
+        all_presets: Dict[str, Optional[int]] = {PRESET_NONE: None}
+        if self.info.primary_value:
```

Comment:
We shouldn't need this check since primary value isn't optional.

Reply:
Thanks missed that one in my sweep